src/main/java/core/basesyntax/dao/figure/FigureDaoImpl.java

```diff
@@ -12,11 +18,37 @@ public FigureDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public T save(T figure) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(figure);
+            transaction.commit();
+            return figure;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Could not save Figure: " + figure, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public List<T> findByColor(String color, Class<T> clazz) {
-        return null;
+        try (Session session = sessionFactory.openSession()) {
+            CriteriaBuilder builder = session.getCriteriaBuilder();
+            CriteriaQuery<T> query = builder.createQuery(clazz);
+            Root<T> root = query.from(clazz);
+            Predicate condition = builder.equal(root.get("color"), color);
+            query.select(root).where(condition);
+            return session.createQuery(query).getResultList();
```

Comment:
```suggestion
            return session.createQuery("from " + clazz.getSimpleName() + " f "
                        + "where f.color = :color", clazz)
                        .setParameter("color", color)
                        .getResultList();
```

Reply:
criteria query is for something different (when you have different amount of parameters)

Reply:
Is this such a big problem?  In our previous homeworks we used it for this purpose as well.

Reply:
ok, I agree with you))